core-plugins/src/main/java/co/cask/hydrator/plugin/transform/LogParserTransform.java

```diff
@@ -102,6 +102,15 @@ public void configurePipeline(PipelineConfigurer pipelineConfigurer) throws Ille
       LOG.error("Log format not currently supported.");
       throw new IllegalStateException("Unsupported log format: " + config.logFormat);
     }
+    Schema inputSchema = pipelineConfigurer.getStageConfigurer().getInputSchema();
+    if (inputSchema != null) {
+      if (!validateInputSchemaType(inputSchema.getType())) {
```

Comment:
we expect all input schemas to be of type RECORD. This should be checking the type of the field as configured by config.inputName.